proto/desmos/posts/v1/models.proto

```diff
@@ -0,0 +1,257 @@
+
+  // Index of the character inside the text at which the URL starts
+  uint64 start = 1;
+
+  // Index of the character inside the text at which the URL ends
+  uint64 end = 2;
+
+  // Value of the URL where the user should be redirected to
+  string url = 3;
+
+  // (optional) Display value of the URL
+  string display_url = 4;
+}
+
+// Attachment contains the data of a single post attachment
+message Attachment {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
+  // Id of the subspace inside which the post to which this attachment should be
+  // connected is
+  uint64 subspace_id = 1 [ (gogoproto.customname) = "SubspaceID" ];
+
+  // Id of the post to which this attachment should be connected
+  uint64 post_id = 2 [ (gogoproto.customname) = "PostID" ];
+
+  // If of this attachment
+  uint32 id = 3 [ (gogoproto.customname) = "ID" ];
+
+  // Content of the attachment
+  google.protobuf.Any content = 4;
+}
+
+// Media represents a media attachment
+message Media {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
+  string uri = 2;
+  string mime_type = 3;
+}
+
+// Poll represents a poll attachment
+message Poll {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
+  // Question of the poll
+  string question = 1;
+
+  // Answers the users can choose from
+  repeated ProvidedAnswer provided_answers = 2 [ (gogoproto.nullable) = false ];
+
+  // Date at which the poll will close
+  google.protobuf.Timestamp end_date = 3
+      [ (gogoproto.nullable) = false, (gogoproto.stdtime) = true ];
+
+  // Whether the poll allows multiple choices from the same user or not
+  bool allows_multiple_answers = 4;
+
+  // Whether the poll allows to edit an answer or not
+  bool allows_answer_edits = 5;
+
+  // Final poll results
+  PollTallyResults final_tally_results = 6;
+
+  // Provided answer contains the details of a possible poll answer
+  message ProvidedAnswer {
+    option (gogoproto.equal) = true;
+    option (gogoproto.goproto_stringer) = true;
+
+    // (optional) Text of the answer
+    string text = 1;
+
+    // Attachments of the answer
+    repeated Attachment attachments = 2 [ (gogoproto.nullable) = false ];
+  }
+}
+
+// UserAnswer represents a user answer to a poll
+message UserAnswer {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
+  // Subspace id inside which the post related to this attachment is located
+  uint64 subspace_id = 1 [ (gogoproto.customname) = "SubspaceID" ];
+
+  // Id of the post associated to this attachment
+  uint64 post_id = 2 [ (gogoproto.customname) = "PostID" ];
+
+  // Id of the poll to which this answer is associated
+  uint32 poll_id = 3 [ (gogoproto.customname) = "PollID" ];
+
+  // Indexes of the answers inside the ProvidedAnswers array
+  repeated uint32 answers_indexes = 4;
+
+  // Address of the user answering the poll
+  bytes user = 5 [ (gogoproto.casttype) =
```

Comment:
Why is the type of user is `AccAddress` instead of bech32 prefixed string? Is it to prevent the double vote with the same address via ibc or something else?

Reply:
Reverted to use a `string` instead of a `bytes` representation